src/figuresClasses.ts

```diff
@@ -1,11 +1,95 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0) {
+      throw new Error(`Side 'a' must be greater than 0, but got ${a}`);
+    }
+
+    if (b <= 0) {
+      throw new Error(`Side 'b' must be greater than 0, but got ${b}`);
+    }
+
+    if (c <= 0) {
+      throw new Error(`Side 'c' must be greater than 0, but got ${c}`);
+    }
+
+    if (!this.isValidTriangle(a, b, c)) {
+      throw new Error("This triangle can't exist.");
+    }
+  }
+
+  private isValidTriangle(a: number, b: number, c: number): boolean {
+    return a + b > c && a + c > b && b + c > a;
+  }
+
+  public getArea(): number {
```

Comment:
The 'getArea' method should not accept any parameters as all necessary values are already available as object fields. Please remove the parameters from 'isValidTriangle' method call within the constructor.